### ISSUE TITLE
Support for UTF16BE encoded strings

### ISSUE THREAD

**Issue by NateZimmer** [fh1ch/node-bacstack#59]
If i'm not mistaken, a decent amount of bacnet devices out there encode strings via UTF16BE where as node only supports UTF16LE(little endian). Parser craps itself where as YABE & VTS handle such devices just fine.  
The string encoding switch in ``bacnet-asn1.js`` @ ~800 ``multiCharsetCharacterstringDecode`` doesn't detect the UTF16BE case , then just returns a null string. Once it hits the ``result.len`` this generates an unhanded exception and crashes. 

Edit: NPM pulled down version 1-beta.9 , will try 10.

**Comment by NateZimmer** [comment]
Yup... even with the latest, can't detect UTF16BE. 
e.g. can't decode the following: 
`` 005a004e0054 `` --> ZNT

**Comment by fh1ch** [comment]
Hi @NateZimmer 

Thanks for creating a bug report and providing all those inputs 👍 

The string encoding in node-bacstack was implemented quiet opportunistic (as well as many other things) due to two reasons:
- String encoding in BACNET is a mess (no official names and versions are mentioned in the official ASHRE BACNET standard).
- Node.JS doesn't support string encodings too well (that's why we added `iconv-lite` in #35).

Furthermore, according to the standard only UCS2 (UTF16LE) is officially specified (not sure why many devices use UTF16BE tough).

However, I tried to come up with a solution and think that I at least fixed the exception in PR #60 and additionally implemented more encodings. Since I don't really have a setup to verify this patch, would you mind giving it a try and provide feedback?

Many thanks in advance!

Cheers

**Comment by NateZimmer** [comment]
Hey, thanks for the quick response. Sorry get back late on testing this. Its a bit tedious since I had to update my code to account for other changes. 
Regarding the following in ``bacnet-asn`.js``
```javascript
    case baEnum.BacnetCharacterStringEncodings.CHARACTER_UCS2:
      if (decodingBuffer && ((decodingBuffer[decodingOffset] === 0xFE) && (decodingBuffer[decodingOffset + 1] === 0xFF))) {
        return 'UTF-16BE';
      }
```
So sadly this didn't work because the string isn't encoded with ``0xFF 0xFE``(LE) or ``0xFE 0xFF``(BE) to indicate its endianess. 

According to wikipedia: 
```
If the BOM is missing, RFC 2781 says that big-endian encoding should be assumed.
```
Consequently I would recommend: 
```javascript
       if (decodingBuffer && ((decodingBuffer[decodingOffset] === 0xFE) && (decodingBuffer[decodingOffset + 1] === 0xFF)) || (decodingBuffer[decodingOffset]==0) ) {
        return 'UTF-16BE';
      }
```
^ The following fixed my encoding issues

**Comment by fh1ch** [comment]
Hi @NateZimmer 

Thanks for feedback on this. I adapted the MR, so BE is now default for the UTF-16 encoding as it's handled by other BACNET stacks as well.

All of those changes are available in the freshly released `0.0.1-beta.11` version.

Let me know if somsething doesn't work as expected...

Cheers
